src/mono/sample/wasm/browser/main.js

```diff
@@ -24,29 +7,35 @@ function add(a, b) {
 function sub(a, b) {
     return a - b;
 }
+
 try {
-    const { runtimeBuildInfo, setModuleImports, getAssemblyExports, runMain } = await createDotnetRuntimeTyped({
-        configSrc: "./mono-config.json",
-        onConfigLoaded: (config) => {
-            // This is called during emscripten `dotnet.wasm` instantiation, after we fetched config.
-            console.log('user code Module.onConfigLoaded');
-            // config is loaded and could be tweaked before the rest of the runtime startup sequence
-            config.environmentVariables["MONO_LOG_LEVEL"] = "debug"
-        },
-        preInit: () => { console.log('user code Module.preInit'); },
-        preRun: () => { console.log('user code Module.preRun'); },
-        onRuntimeInitialized: () => {
-            console.log('user code Module.onRuntimeInitialized');
-            // here we could use API passed into this callback
-            // Module.FS.chdir("/");
-        },
-        onDotnetReady: () => {
-            // This is called after all assets are loaded.
-            console.log('user code Module.onDotnetReady');
-        },
-        postRun: () => { console.log('user code Module.postRun'); },
-    });
-    
+    const { runtimeBuildInfo, setModuleImports, getAssemblyExports, runMain, getConfig } = await dotnet
+        .withConsoleForwarding()
+        .withElementOnExit()
+        .withModuleConfig({
```

Comment:
why are we configuring some things with a literate '.withXxx' api here and then passing in a big config dictionary?

Reply:
It's just for backward compatibility. We don't allow everything with those `with*` methods. Also this method is not "public" as it is not in the typescript public definition.

Reply:
We actually don't need any of those callbacks now. They could hr tlremoved from the sample. It's just showing that they are still there and working, in case somebody needs to go low level.

This could be further improved later and it should not block the PR, I think

Reply:
WebHostBuilder* seems to go with Use* or Configure* rather than With*?